In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
df = pd.read_csv("../data/raw/student-mat.csv", sep=";")

In [4]:
df.head()

,school,sex,age,address,famsize,Pstatus,Medu,Fedu,Mjob,Fjob,...,famrel,freetime,goout,Dalc,Walc,health,absences,G1,G2,G3
0,GP,F,18,U,GT3,A,4,4,at_home,teacher,...,4,3,4,1,1,3,6,5,6,6
1,GP,F,17,U,GT3,T,1,1,at_home,other,...,5,3,3,1,1,3,4,5,5,6
2,GP,F,15,U,LE3,T,1,1,at_home,other,...,4,3,2,2,3,3,10,7,8,10
3,GP,F,15,U,GT3,T,4,2,health,services,...,3,2,2,1,1,5,2,15,14,15
4,GP,F,16,U,GT3,T,3,3,other,other,...,4,3,2,1,2,5,4,6,10,10


In [5]:
df.shape

(395, 33)

### Observação

O dataset possui 395 registros e 33 variáveis. Cada registro representa um estudante, enquanto as colunas representam diferentes características demográficas, sociais, escolares e de desempenho.

In [6]:
df.columns

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='str')

### Variáveis disponíveis

O dataset contém 33 variáveis relacionadas a características demográficas, familiares, escolares, hábitos, contexto social e desempenho acadêmico dos estudantes.

A identificação inicial das variáveis permite compreender quais dimensões podem ser exploradas posteriormente. A classificação estatística de cada variável será realizada após a consulta à documentação do dataset e à análise de seus valores e tipos.

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   school      395 non-null    str  
 1   sex         395 non-null    str  
 2   age         395 non-null    int64
 3   address     395 non-null    str  
 4   famsize     395 non-null    str  
 5   Pstatus     395 non-null    str  
 6   Medu        395 non-null    int64
 7   Fedu        395 non-null    int64
 8   Mjob        395 non-null    str  
 9   Fjob        395 non-null    str  
 10  reason      395 non-null    str  
 11  guardian    395 non-null    str  
 12  traveltime  395 non-null    int64
 13  studytime   395 non-null    int64
 14  failures    395 non-null    int64
 15  schoolsup   395 non-null    str  
 16  famsup      395 non-null    str  
 17  paid        395 non-null    str  
 18  activities  395 non-null    str  
 19  nursery     395 non-null    str  
 20  higher      395 non-null    str  
 21  inte

Estrutura dos dados: o dataset é composto por 395 registros e 33 variáveis. Todas as colunas apresentam 395 valores não nulos, indicando ausência de valores ausentes identificados pelo Pandas nesta etapa inicial. Quanto aos tipos computacionais, o conjunto possui 16 variáveis armazenadas como números inteiros (int64) e 17 variáveis armazenadas como texto (str), demonstrando a presença de informações numéricas e categóricas no mesmo dataset.

In [8]:
# parte de ver as estatisticas descritivas dessas variaveis 
df.describe()

,age,Medu,Fedu,traveltime,studytime,failures,famrel,freetime,goout,Dalc,Walc,health,absences,G1,G2,G3
count,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000,395.000000
mean,16.696203,2.749367,2.521519,1.448101,2.035443,0.334177,3.944304,3.235443,3.108861,1.481013,2.291139,3.554430,5.708861,10.908861,10.713924,10.415190
std,1.276043,1.094735,1.088201,0.697505,0.839240,0.743651,0.896659,0.998862,1.113278,0.890741,1.287897,1.390303,8.003096,3.319195,3.761505,4.581443
min,15.000000,0.000000,0.000000,1.000000,1.000000,0.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,0.000000,3.000000,0.000000,0.000000
25%,16.000000,2.000000,2.000000,1.000000,1.000000,0.000000,4.000000,3.000000,2.000000,1.000000,1.000000,3.000000,0.000000,8.000000,9.000000,8.000000
50%,17.000000,3.000000,2.000000,1.000000,2.000000,0.000000,4.000000,3.000000,3.000000,1.000000,2.000000,4.000000,4.000000,11.000000,11.000000,11.000000
75%,18.000000,4.000000,3.000000,2.000000,2.000000,0.000000,5.000000,4.000000,4.000000,2.000000,3.000000,5.000000,8.000000,13.000000,13.000000,14.000000
max,22.000000,4.000000,4.000000,4.000000,4.000000,3.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,75.000000,19.000000,19.000000,20.000000


### Estatísticas descritivas iniciais

A análise descritiva inicial indica que a idade dos estudantes está concentrada principalmente entre 16 e 18 anos, com média de aproximadamente 16,7 anos e mediana de 17 anos.

As médias das notas apresentam redução entre G1 (10,91), G2 (10,71) e G3 (10,42), enquanto a mediana permanece em 11 nas três avaliações. Essa diferença deverá ser investigada posteriormente por meio da análise da distribuição das notas.

A variável `absences` apresenta grande dispersão, com mediana de 4 faltas e máximo de 75. Essa diferença sugere uma possível assimetria e a presença de valores extremos, que deverão ser investigados nas próximas etapas.

Algumas variáveis numéricas são, na realidade, códigos ou escalas ordinais. Portanto, seus valores não serão interpretados quantitativamente antes da consulta à documentação oficial do dataset.





In [9]:
#investigando valores ausentes
df.isnull().sum()

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
failures      0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Dalc          0
Walc          0
health        0
absences      0
G1            0
G2            0
G3            0
dtype: int64

Não foram identificados valores ausentes nas 33 variáveis analisadas. Todas as colunas apresentam 395 observações preenchidas.

In [10]:
#vendo as duplicidades
df.duplicated().sum()

np.int64(0)

### Verificação inicial da qualidade dos dados

A análise inicial da qualidade dos dados não identificou valores ausentes nas 33 variáveis analisadas. Também não foram encontrados registros completamente duplicados entre as 395 observações.

Esses resultados indicam que não há, nesta etapa, necessidade de tratamento de valores nulos ou remoção de duplicidades completas. Entretanto, a qualidade dos dados ainda deverá ser investigada quanto a possíveis valores inválidos, inconsistências e valores extremos.

# 3. Dicionário de Dados

Esta seção apresenta a definição das variáveis utilizadas no dataset, suas respectivas codificações e sua classificação estatística.

As definições foram obtidas a partir da documentação oficial do dataset Student Performance, disponibilizada pelo UCI Machine Learning Repository.

| Variável  | Descrição                       | Tipo estatístico           | Valores/Codificação                             | Observação analítica        |
| --------- | ------------------------------- | -------------------------- | ----------------------------------------------- | --------------------------- |
| `school`  | Escola do estudante             | Categórica nominal         | GP = Gabriel Pereira; MS = Mousinho da Silveira | Categorias sem ordem        |
| `sex`     | Sexo do estudante               | Categórica nominal/binária | F = feminino; M = masculino                     | Duas categorias             |
| `age`     | Idade do estudante              | Quantitativa discreta      | 15–22 anos                                      | Valor numérico              |
| `address` | Tipo de endereço residencial    | Categórica nominal/binária | U = urbano; R = rural                           | Categorias sem ordem        |
| `famsize` | Tamanho da família              | Categórica nominal/binária | LE3 = ≤3; GT3 = >3                              | Dataset já agrupa o tamanho |
| `Pstatus` | Situação de coabitação dos pais | Categórica nominal/binária | T = juntos; A = separados                       | Categorias sem ordem        |


👨‍👩‍👧 Bloco 2 — Contexto familiar

| Variável   | Descrição                  | Tipo estatístico   | Valores / Codificação                                                                            | Observação analítica                 |
| ---------- | -------------------------- | ------------------ | ------------------------------------------------------------------------------------------------ | ------------------------------------ |
| `Medu`     | Escolaridade da mãe        | Categórica ordinal | 0 = nenhuma; 1 = ensino fundamental 4º ano; 2 = 5º–9º ano; 3 = ensino médio; 4 = ensino superior | As categorias possuem ordem          |
| `Fedu`     | Escolaridade do pai        | Categórica ordinal | 0 = nenhuma; 1 = ensino fundamental 4º ano; 2 = 5º–9º ano; 3 = ensino médio; 4 = ensino superior | As categorias possuem ordem          |
| `Mjob`     | Trabalho/profissão da mãe  | Categórica nominal | teacher, health, services, at_home, other                                                        | Não existe ordem entre as categorias |
| `Fjob`     | Trabalho/profissão do pai  | Categórica nominal | teacher, health, services, at_home, other                                                        | Não existe ordem entre as categorias |
| `guardian` | Responsável pelo estudante | Categórica nominal | mother, father, other                                                                            | Categorias sem ordem                 |


📚 Bloco 3 — Escola e hábitos de estudo

| Variável     | Descrição                                 | Tipo estatístico           | Valores / Codificação                                | Observação analítica                                   |
| ------------ | ----------------------------------------- | -------------------------- | ---------------------------------------------------- | ------------------------------------------------------ |
| `reason`     | Motivo para escolha da escola             | Categórica nominal         | home, reputation, course, other                      | Categorias sem ordem                                   |
| `traveltime` | Tempo de deslocamento entre casa e escola | Categórica ordinal         | 1 = <15 min; 2 = 15–30 min; 3 = 30 min–1 h; 4 = >1 h | Categorias possuem ordem                               |
| `studytime`  | Tempo semanal de estudo                   | Categórica ordinal         | 1 = <2 h; 2 = 2–5 h; 3 = 5–10 h; 4 = >10 h           | Categorias possuem ordem                               |
| `failures`   | Número de reprovações anteriores          | Quantitativa discreta      | 0–3; valores ≥4 são representados como 4             | É uma contagem, mas possui agrupamento no valor máximo |
| `schoolsup`  | Apoio educacional adicional               | Categórica nominal/binária | yes, no                                              | Duas categorias                                        |
| `famsup`     | Apoio educacional da família              | Categórica nominal/binária | yes, no                                              | Duas categorias                                        |
| `paid`       | Aulas particulares pagas na disciplina    | Categórica nominal/binária | yes, no                                              | Duas categorias                                        |


🏃 Bloco 4 — Atividades e contexto pessoal

| Variável     | Descrição                                    | Tipo estatístico           | Valores / Codificação | Observação analítica |
| ------------ | -------------------------------------------- | -------------------------- | --------------------- | -------------------- |
| `activities` | Participação em atividades extracurriculares | Categórica nominal/binária | yes, no               | Duas categorias      |
| `nursery`    | Frequência anterior à escola infantil        | Categórica nominal/binária | yes, no               | Duas categorias      |
| `higher`     | Desejo de cursar ensino superior             | Categórica nominal/binária | yes, no               | Duas categorias      |
| `internet`   | Acesso à Internet em casa                    | Categórica nominal/binária | yes, no               | Duas categorias      |
| `romantic`   | Possui relacionamento romântico              | Categórica nominal/binária | yes, no               | Duas categorias      |


🧠 Bloco 5 — Escalas comportamentais e sociais

| Variável   | Descrição                                 | Tipo estatístico   | Escala                           | Observação analítica |
| ---------- | ----------------------------------------- | ------------------ | -------------------------------- | -------------------- |
| `famrel`   | Qualidade das relações familiares         | Categórica ordinal | 1 = muito ruim → 5 = excelente   | Escala ordenada      |
| `freetime` | Tempo livre após a escola                 | Categórica ordinal | 1 = muito baixo → 5 = muito alto | Escala ordenada      |
| `goout`    | Frequência de saída com amigos            | Categórica ordinal | 1 = muito baixa → 5 = muito alta | Escala ordenada      |
| `Dalc`     | Consumo de álcool durante a semana        | Categórica ordinal | 1 = muito baixo → 5 = muito alto | Escala ordenada      |
| `Walc`     | Consumo de álcool durante o fim de semana | Categórica ordinal | 1 = muito baixo → 5 = muito alto | Escala ordenada      |
| `health`   | Estado de saúde atual                     | Categórica ordinal | 1 = muito ruim → 5 = muito bom   | Escala ordenada      |


📊 Bloco 6 — Desempenho acadêmico

| Variável   | Descrição                  | Tipo estatístico      | Valores / Escala | Observação analítica                 |
| ---------- | -------------------------- | --------------------- | ---------------- | ------------------------------------ |
| `absences` | Número de faltas escolares | Quantitativa discreta | 0–75 no dataset  | Contagem de faltas                   |
| `G1`       | Nota do primeiro período   | Quantitativa discreta | 0–20             | Nota acadêmica                       |
| `G2`       | Nota do segundo período    | Quantitativa discreta | 0–20             | Nota acadêmica                       |
| `G3`       | Nota final                 | Quantitativa discreta | 0–20             | Principal medida de desempenho final |


In [11]:
#frequencias de categorias-tipos de dados e distribuicao
df["sex"].value_counts()

sex
F    208
M    187
Name: count, dtype: int64

In [12]:
df["sex"].value_counts(normalize=True).mul(100).round(2)

sex
F    52.66
M    47.34
Name: proportion, dtype: float64

### Insight — variável `sex`

A amostra possui 395 estudantes, sendo 208 classificados como F (52,66%) e 187 como M (47,34%). Observa-se uma pequena predominância da categoria F, porém a distribuição entre as duas categorias é relativamente equilibrada.

Essa análise descreve a composição da amostra e não permite, isoladamente, estabelecer relação entre `sex` e desempenho acadêmico. Para investigar possíveis diferenças de desempenho entre os grupos, seria necessária uma análise posterior envolvendo as variáveis de desempenho, como `G3`.

In [13]:
df["school"].value_counts()

school
GP    349
MS     46
Name: count, dtype: int64

In [14]:
df["school"].value_counts(normalize=True).mul(100).round(2)

school
GP    88.35
MS    11.65
Name: proportion, dtype: float64

### Insight — variável `school`

A amostra é composta por 395 estudantes, sendo 349 (88,35%) da escola GP e 46 (11,65%) da escola MS. Observa-se, portanto, uma forte concentração das observações na escola GP.

Essa distribuição deve ser considerada em análises posteriores que comparem as escolas, especialmente análises relacionadas ao desempenho acadêmico, pois os grupos apresentam tamanhos amostrais bastante diferentes.

A distribuição da variável `school` é categórica nominal, sendo o gráfico de barras uma visualização adequada para comparar a frequência das duas categorias.

In [15]:
df["age"].value_counts().sort_index()

age
15     82
16    104
17     98
18     82
19     24
20      3
21      1
22      1
Name: count, dtype: int64

In [16]:
df["age"].value_counts(normalize=True).mul(100).round(2).sort_index()

age
15    20.76
16    26.33
17    24.81
18    20.76
19     6.08
20     0.76
21     0.25
22     0.25
Name: proportion, dtype: float64

### Insight — variável `age`

A distribuição etária da amostra concentra-se principalmente entre 15 e 18 anos. Essa faixa reúne 366 dos 395 estudantes, correspondendo a aproximadamente 92,66% da amostra.

A idade de 16 anos apresenta a maior frequência, com 104 estudantes (26,33%), seguida por 17 anos, com 98 estudantes (24,81%).

As idades de 20 a 22 anos apresentam frequência muito baixa, totalizando apenas 5 estudantes. Dessa forma, a amostra é predominantemente composta por estudantes entre 15 e 18 anos.

Essa distribuição é consistente com a média de 16,70 anos e a mediana de 17 anos observadas na análise descritiva.

In [21]:
categorical_columns = [
    "school",
    "sex",
    "address",
    "famsize",
    "Pstatus",
    "Mjob",
    "Fjob",
    "reason",
    "guardian",
    "schoolsup",
    "famsup",
    "paid",
    "activities",
    "nursery",
    "higher",
    "internet",
    "romantic"
]

In [18]:
for column in categorical_columns:
    print(f"\n{column}")
    print(df[column].unique())


school
<StringArray>
['GP', 'MS']
Length: 2, dtype: str

sex
<StringArray>
['F', 'M']
Length: 2, dtype: str

address
<StringArray>
['U', 'R']
Length: 2, dtype: str

famsize
<StringArray>
['GT3', 'LE3']
Length: 2, dtype: str

Pstatus
<StringArray>
['A', 'T']
Length: 2, dtype: str

Mjob
<StringArray>
['at_home', 'health', 'other', 'services', 'teacher']
Length: 5, dtype: str

Fjob
<StringArray>
['teacher', 'other', 'services', 'health', 'at_home']
Length: 5, dtype: str

reason
<StringArray>
['course', 'other', 'home', 'reputation']
Length: 4, dtype: str

guardian
<StringArray>
['mother', 'father', 'other']
Length: 3, dtype: str

schoolsup
<StringArray>
['yes', 'no']
Length: 2, dtype: str

famsup
<StringArray>
['no', 'yes']
Length: 2, dtype: str

paid
<StringArray>
['no', 'yes']
Length: 2, dtype: str

activities
<StringArray>
['no', 'yes']
Length: 2, dtype: str

nursery
<StringArray>
['yes', 'no']
Length: 2, dtype: str

higher
<StringArray>
['yes', 'no']
Length: 2, dtype: str

internet
<

In [19]:
for column in categorical_columns:
    print(f"\n{column}")
    print(df[column].value_counts())


school
school
GP    349
MS     46
Name: count, dtype: int64

sex
sex
F    208
M    187
Name: count, dtype: int64

address
address
U    307
R     88
Name: count, dtype: int64

famsize
famsize
GT3    281
LE3    114
Name: count, dtype: int64

Pstatus
Pstatus
T    354
A     41
Name: count, dtype: int64

Mjob
Mjob
other       141
services    103
at_home      59
teacher      58
health       34
Name: count, dtype: int64

Fjob
Fjob
other       217
services    111
teacher      29
at_home      20
health       18
Name: count, dtype: int64

reason
reason
course        145
home          109
reputation    105
other          36
Name: count, dtype: int64

guardian
guardian
mother    273
father     90
other      32
Name: count, dtype: int64

schoolsup
schoolsup
no     344
yes     51
Name: count, dtype: int64

famsup
famsup
yes    242
no     153
Name: count, dtype: int64

paid
paid
no     214
yes    181
Name: count, dtype: int64

activities
activities
yes    201
no     194
Name: count, dtype: int64

nu

In [22]:
for column in categorical_columns:
    total = df[column].value_counts().sum()
    print(f"{column}: {total} registros")

school: 395 registros
sex: 395 registros
address: 395 registros
famsize: 395 registros
Pstatus: 395 registros
Mjob: 395 registros
Fjob: 395 registros
reason: 395 registros
guardian: 395 registros
schoolsup: 395 registros
famsup: 395 registros
paid: 395 registros
activities: 395 registros
nursery: 395 registros
higher: 395 registros
internet: 395 registros
romantic: 395 registros


# Validação das variáveis numéricas e ordinais

In [23]:
numeric_columns = [
    "age",
    "Medu",
    "Fedu",
    "traveltime",
    "studytime",
    "failures",
    "famrel",
    "freetime",
    "goout",
    "Dalc",
    "Walc",
    "health",
    "absences",
    "G1",
    "G2",
    "G3"
]

In [24]:
for column in numeric_columns:
    print(
        f"{column}: "
        f"mínimo={df[column].min()}, "
        f"máximo={df[column].max()}"
    )

age: mínimo=15, máximo=22
Medu: mínimo=0, máximo=4
Fedu: mínimo=0, máximo=4
traveltime: mínimo=1, máximo=4
studytime: mínimo=1, máximo=4
failures: mínimo=0, máximo=3
famrel: mínimo=1, máximo=5
freetime: mínimo=1, máximo=5
goout: mínimo=1, máximo=5
Dalc: mínimo=1, máximo=5
Walc: mínimo=1, máximo=5
health: mínimo=1, máximo=5
absences: mínimo=0, máximo=75
G1: mínimo=3, máximo=19
G2: mínimo=0, máximo=19
G3: mínimo=0, máximo=20


In [27]:
for column in numeric_columns:
    print(
        f"{column}: "
        f"mínimo={df[column].min()} | "
        f"máximo={df[column].max()} | "
        f"valores distintos={df[column].nunique()}"
    )

age: mínimo=15 | máximo=22 | valores distintos=8
Medu: mínimo=0 | máximo=4 | valores distintos=5
Fedu: mínimo=0 | máximo=4 | valores distintos=5
traveltime: mínimo=1 | máximo=4 | valores distintos=4
studytime: mínimo=1 | máximo=4 | valores distintos=4
failures: mínimo=0 | máximo=3 | valores distintos=4
famrel: mínimo=1 | máximo=5 | valores distintos=5
freetime: mínimo=1 | máximo=5 | valores distintos=5
goout: mínimo=1 | máximo=5 | valores distintos=5
Dalc: mínimo=1 | máximo=5 | valores distintos=5
Walc: mínimo=1 | máximo=5 | valores distintos=5
health: mínimo=1 | máximo=5 | valores distintos=5
absences: mínimo=0 | máximo=75 | valores distintos=34
G1: mínimo=3 | máximo=19 | valores distintos=17
G2: mínimo=0 | máximo=19 | valores distintos=17
G3: mínimo=0 | máximo=20 | valores distintos=18


As variáveis numéricas e ordinais foram validadas por meio da verificação dos valores mínimos, máximos e da quantidade de valores distintos. Os intervalos observados são compatíveis com a documentação oficial do dataset, não sendo identificados valores fora do domínio esperado durante a inspeção inicial.

In [35]:
expected_ranges = {
    "age": (15, 22),
    "Medu": (0, 4),
    "Fedu": (0, 4),
    "traveltime": (1, 4),
    "studytime": (1, 4),
    "failures": (0, 3),
    "famrel": (1, 5),
    "freetime": (1, 5),
    "goout": (1, 5),
    "Dalc": (1, 5),
    "Walc": (1, 5),
    "health": (1, 5),
    "absences": (0, 75),
    "G1": (0, 20),
    "G2": (0, 20),
    "G3": (0, 20)
}

In [36]:
type(expected_ranges)

dict

In [31]:
expected_ranges = {...}

In [37]:
for column, (min_expected, max_expected) in expected_ranges.items():

    invalid_values = df[
        (df[column] < min_expected) |
        (df[column] > max_expected)
    ]

    print(
        f"{column}: "
        f"{len(invalid_values)} valores fora do domínio"
    )

age: 0 valores fora do domínio
Medu: 0 valores fora do domínio
Fedu: 0 valores fora do domínio
traveltime: 0 valores fora do domínio
studytime: 0 valores fora do domínio
failures: 0 valores fora do domínio
famrel: 0 valores fora do domínio
freetime: 0 valores fora do domínio
goout: 0 valores fora do domínio
Dalc: 0 valores fora do domínio
Walc: 0 valores fora do domínio
health: 0 valores fora do domínio
absences: 0 valores fora do domínio
G1: 0 valores fora do domínio
G2: 0 valores fora do domínio
G3: 0 valores fora do domínio


# Conclusão da coleta e validação inicial

## Situação da base

A base utilizada neste projeto possui 395 registros e 33 variáveis.

Durante a etapa de coleta e inspeção inicial foram realizadas verificações relacionadas à estrutura, dimensões, tipos de dados, estatísticas descritivas, valores ausentes, registros duplicados, categorias, frequências e domínios das variáveis.

Não foram identificados valores ausentes ou registros duplicados na inspeção inicial.

As variáveis categóricas foram avaliadas quanto às categorias observadas e à distribuição dos registros.

As variáveis numéricas e ordinais foram avaliadas quanto aos valores mínimos, máximos e quantidade de valores distintos. Também foi realizada uma validação dos domínios esperados.

A validação de domínio não identificou valores fora dos intervalos definidos para as variáveis avaliadas.

A etapa de coleta e validação inicial foi concluída, e a base está pronta para a próxima etapa do projeto: limpeza e preparação dos dados para análise.
